### PR TITLE
Donation form test 01

### PR DESCRIPTION
**Pull request by Abban**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
 		"wmde/email-address": "~1.0",
 		"wmde/euro": "~1.0",
 		"wmde/clock": "~1.0",
-		"wmde/fundraising-donations": "~14.0",
+		"wmde/fundraising-donations": "dev-test",
```
Comment (gbirke) [inline composer.json:19]: Interesting, shouldn't that be `dev-company-with-contact`?

File: app/Controllers/Donation/AddDonationController.php [hunk 2]
```diff
@@ -63,6 +63,10 @@ private function newHttpResponse( AddDonationResponse $responseModel ): Response
 	private function createDonationRequest( Request $request ): AddDonationRequest {
 		$donationRequest = new AddDonationRequest();
 
+		if ( $request->get( 'addressType', '' ) === 'company_with_contact' ) {
+			$request->query->set( 'addressType', AddressType::LEGACY_COMPANY );
+		}
+
```
Comment (gbirke) [inline app/Controllers/Donation/AddDonationController.php:69]: This hack should go into `getSafeDonorType` IMHO, with a comment or maybe even with a restructuring of that code so we know what's going on.

File: app/Controllers/Validation/ValidateAddressController.php [hunk 3]
```diff
@@ -24,11 +24,16 @@ class ValidateAddressController {
...
+		$isCompanyWithContact = $this->handleCompanyWithContact( $request );
+
 		$addressType = $this->getAddressType( $request );
 		if ( $addressType === AddressType::PERSON ) {
 			$nameViolations = $this->getPersonViolations( $request );
 		} elseif ( $addressType === AddressType::COMPANY ) {
 			$nameViolations = $this->getCompanyViolations( $request );
+			if ( $isCompanyWithContact ) {
+				$nameViolations = array_merge( $nameViolations, $this->getPersonViolations( $request ) );
+			}
```
Comment (gbirke) [inline app/Controllers/Validation/ValidateAddressController.php:36]: This check should go into `getCompanyViolations` to avoid 2 levels of nesting here

**Review by gbirke — CHANGES_REQUESTED** [review]
If we quickly need to deploy this to make the test work, you can consider this as approved. But we then should address my comments as a follow-up

**Review by gbirke — APPROVED** [review]

**Comment by moiikana** [comment]
needs to point to fundraising-donations v14.1

**Review by moiikana — APPROVED** [review]

**Pull request merged**